### DISCUSSION TITLE
Want to capture full spectrum of contributions but foiled by rate and return limits of API calls

### DISCUSSION THREAD

**Discussion by heidiguenin** [community/community#24880] [Apps, API and Webhooks]
<p>Trying to understand the full breadth of community contributions - cumulative and by specific time period - to a repo. <a href="https://github.com/HeidiMG/git-thanks" rel="nofollow noopener">We built the beginning of a script</a>, but quickly ran into the fact that we could only get 30 results per call. Sounds like that might be a pagination issue, but looks like 1000 is the limit on query results anyway. To be able to understand the cumulative contributions, we’ll need to be able to retrieve *all* of the results for the history of the repo.</p>
<p>Any ideas for how to get around this?</p>
<p>It would be great if the GitHub API Statistics call could include comments and other contributions in addition to commits.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/heidimg">@heidimg</a>,</p>
<p>Thank you for being here! </p>
<blockquote>
<p>but looks like 1000 is the limit on query results anyway. To be able to understand the cumulative contributions, we’ll need to be able to retrieve *all* of the results for the history of the repo.</p>
</blockquote>
<blockquote>
<blockquote>
<hr>
</blockquote>
</blockquote>
<p>Yes that is correct, that limit is documented here: <a href="https://developer.github.com/v3/search/#search" rel="nofollow noopener">https://developer.github.com/v3/search/#search</a></p>
<p>You’ll also see the same limit in the web UI – you can’t fetch more than 1000 results via Search.</p>
<p>The filters which code search supports are listed here:</p>
<p><a href="https://help.github.com/en/articles/searching-code" rel="nofollow noopener">https://help.github.com/en/articles/searching-code</a></p>
<p>That’s what you can use to try and slice your search query into several smaller queries, run each of them (and hope each will return fewer than 1000 results), and then combine the results so that you have them all.</p>
<p>If you’re unable to get search to give you the results you need, you might consider using an external project like <a href="https://cloud.google.com/blog/products/gcp/github-on-bigquery-analyze-all-the-open-source-code" rel="nofollow noopener">https://cloud.google.com/blog/products/gcp/github-on-bigquery-analyze-all-the-open-source-code</a> to get the data you need.</p>
<p>I hope this helps!</p>
